cmd/openshift/operator/kodata/tekton-addon/addons/02-clustertasks/source_local/buildah/buildah-task.yaml

```diff
@@ -76,7 +77,25 @@ spec:
         -f $(params.DOCKERFILE) -t $(params.IMAGE) $(params.CONTEXT)
 
       [[ "$(params.SKIP_PUSH)" == "true" ]] && echo "Push skipped" && exit 0
-      [[ "$(workspaces.dockerconfig.bound)" == "true" ]] && export DOCKER_CONFIG="$(workspaces.dockerconfig.path)"
+
+      if [[ "$(workspaces.dockerconfig.bound)" == "true" ]]; then
+
+        # if config.json exists at workspace root, we use that
+        if test -f "$(workspaces.dockerconfig.path)/config.json"; then
+          export DOCKER_CONFIG="$(workspaces.dockerconfig.path)"
+
+        # else we look for .dockerconfigjson at the root
+        elif test -f "$(workspaces.dockerconfig.path)/.dockerconfigjson"; then
+          cp "$(workspaces.dockerconfig.path)/.dockerconfigjson" "$HOME/.docker/config.json"
```

Comment:
why do we need to copy, can't we just set this path to DOCKER_CONFIG

Reply:
Because the name is not correct 🙃 `DOCKER_CONFIG` is the path of a folder where `docker` (or `buildah` in that case) will look for a `config.json` file. The file in the workspace is call `.dockerconfigjson` 🙃

Reply:
DOCKER_CONFIG takes a directory which contains a file called `config.json` which is not the case here; we could copy (rename the file) in the workspace itself but we might not have write permissions there; so i chose to copy it to HOME

Reply:
ah, vincent already replied :P

Reply:
ack